ngraph/core/include/openvino/core/visibility.hpp

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// https://gcc.gnu.org/wiki/Visibility
+// Generic helper definitions for shared library support
+#if defined _WIN32 || defined __CYGWIN__
+#    define CORE_HELPER_DLL_IMPORT __declspec(dllimport)
+#    define CORE_HELPER_DLL_EXPORT __declspec(dllexport)
+#    define CORE_HELPER_DLL_LOCAL
+#elif defined(__GNUC__) && __GNUC__ >= 4
+#    define CORE_HELPER_DLL_IMPORT __attribute__((visibility("default")))
+#    define CORE_HELPER_DLL_EXPORT __attribute__((visibility("default")))
+#    define CORE_HELPER_DLL_LOCAL  __attribute__((visibility("hidden")))
+#else
+#    define CORE_HELPER_DLL_IMPORT
+#    define CORE_HELPER_DLL_EXPORT
+#    define CORE_HELPER_DLL_LOCAL
+#endif
```

Comment:
should not we copy all stuff from `ie_api.hpp`?

Reply:
We have some duplication of these structure, I propose to do it separately, the idea of this PR is the creation of entry point for changing nGraph API.